src/core/forward.h

```diff
@@ -204,6 +204,10 @@ static inline int msg_send_buffer(
 			goto error;
 		}
 
+		if(!dst->id) {
+			dst->id = con->id;
+		}
+
```

Comment:
Any more details about why this is needed? I cannot easy relate it to the commit message.

Reply:
This change is needed for siptrace. Below in this file at the line 334 `dst` is passed as an argument to the `SREV_NET_DATA_SENT` related function call.

If the tcp connection ID is assigned to `dst` param then the tcp_connection search is correct and we have a proper siptrace with HAPROXY IP addresses (IP address of client and IP address of HAPROXY) , otherwise tcp connection search is based on real IP addresses (IP addresses of haproxy and IP address of kamailio) and in this case the siptrace contains not the correct IP-addresses.